internal/corazawaf/waf.go

```diff
@@ -39,6 +36,7 @@ type ErrorLogCallback = func(rule types.MatchedRule)
 // All WAF instance fields are immutable, if you update any
 // of them in runtime you might create concurrency issues
 type WAF struct {
+	transactionPool sync.Pool
```

Comment:
This is particularly an important change as different instances of WAF would reuse the same pool of transactions which would end up on reusing the body buffer settings of the first one declared as in https://github.com/M4tteoP/coraza/pull/1/files#diff-b7d81193324b35778d3c6feb2aa00b46357803142e09e27549afcae2c8b47f0cR182